From a8868c41d7b21c1ed148eac81fa04441c6d99ff2 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Thu, 27 Apr 2023 13:17:53 +0800
Subject: [PATCH 17/68] Revert "drm/amdgpu: fix error checking in
 amdgpu_read_mm_registers for soc15"

This reverts commit 4279e87da6c88d63796904184b6f027321e370ab.
---
 drivers/gpu/drm/amd/amdgpu/soc15.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/soc15.c b/drivers/gpu/drm/amd/amdgpu/soc15.c
index 994e6635b8347..7212b9900e0ab 100644
--- a/drivers/gpu/drm/amd/amdgpu/soc15.c
+++ b/drivers/gpu/drm/amd/amdgpu/soc15.c
@@ -382,9 +382,8 @@ static int soc15_read_register(struct amdgpu_device *adev, u32 se_num,
 	*value = 0;
 	for (i = 0; i < ARRAY_SIZE(soc15_allowed_read_registers); i++) {
 		en = &soc15_allowed_read_registers[i];
-		if (!adev->reg_offset[en->hwip][en->inst])
-			continue;
-		else if (reg_offset != (adev->reg_offset[en->hwip][en->inst][en->seg]
+		if (adev->reg_offset[en->hwip][en->inst] &&
+			reg_offset != (adev->reg_offset[en->hwip][en->inst][en->seg]
 					+ en->reg_offset))
 			continue;
 
-- 
2.25.1

